### PR TITLE
OOP Tiny-JS-World

### PR DESCRIPTION
**Pull request by Conversee12**
### oop-tiny-js-world

[Demo](https://conversee12.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/Conversee12/a-tiny-JS-world.git)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by Conversee12** [comment]
I've completed self-checks, did'nt find any common errors

File: submissions/Conversee12/OOP Tiny-Js-World/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+class Inhabitant {
+   constructor(species, name, gender, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+   }
+   get tinyWorld() {
+      return `${this.species}; ${this.name}; ${this.gender}; ${this.saying}`;
```
Comment (OleksiyRudenko) [inline submissions/Conversee12/OOP Tiny-Js-World/index.js:16]: Please employ `Array` methods. Starting array may not necessarily consist of strings. Actually, anything can be an array's element.

File: submissions/Conversee12/OOP Tiny-Js-World/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying) {
+      super('human', name, gender, saying);
+      this.hands = 2;
+      this.legs = 2;
+   }
+   get tinyWorld() {
+      return `${this.species}; ${this.name}; ${this.gender}; ${this.hands}; ${this.legs}; ${this.saying}`;
```
Comment (OleksiyRudenko) [inline submissions/Conversee12/OOP Tiny-Js-World/index.js:27]: Use method overloading.
If we add a new universally shared property, current implementation will require changes to multiple classes' code. The point of encapsulation is that every class handles its own properties and re-uses inherited methods versions to build on top of whatever they provide us with.

Let's disregard the fact that we want to "insert" information about limbs before saying, for simplification.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Conversee12 let's make code more OOP compliant.

**Comment by Conversee12** [comment]
Fixed this. Please, review

**Review by OleksiyRudenko — APPROVED** [review]
@Conversee12 great job done!

**Pull request merged**